### PR TITLE
[runtime] Clean up symbol exports in casting and class introspection.

### PR DESCRIPTION
**Pull request by gparker42**

### CODE REVIEW

**Comment by gparker42** [comment]
@swift-ci please test

**Comment by gparker42** [comment]
@swift-ci please test macos

File: stdlib/public/core/Misc.swift [hunk 1]
```diff
@@ -92,17 +92,14 @@ func _typeName(_ type: Any.Type, qualified: Bool = true) -> String {
     input: UnsafeBufferPointer(start: stringPtr, count: count))
 }
 
-@_inlineable // FIXME(sil-serialize-all)
-@_versioned // FIXME(sil-serialize-all)
-@_silgen_name("swift_getTypeByName")
+@_silgen_name("")
```
Comment (slavapestov) [inline stdlib/public/core/Misc.swift]: What does it mean to have an empty string literal here?

Comment (gparker42) [reply stdlib/public/core/Misc.swift]: It means the function gets the same symbol visibility treatment as ordinary `@_silgen_name` (i.e. internal functions get C-visible hidden symbols), but the mangled name of the function is unchanged. This syntax is a temporary hack and ought to be replaced someday.

The goal is that Swift functions that merely happen to be implemented in C today should have Swift-mangled symbol names.

**Pull request merged**